modules/auxiliary/analyze/crack_linux.rb

```diff
@@ -0,0 +1,276 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/auxiliary/password_cracker'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::PasswordCracker
+
+  def initialize
+    super(
+      'Name'            => 'Password Cracker: Linux',
+      'Description'     => %Q{
+          This module uses John the Ripper or Hashcat to identify weak passwords that have been
+        acquired from unshadowed passwd files from Unix/Linux systems. The module will only crack
+        MD5, BSDi and DES implementations by default. Set Crypt to true to also try to crack
+        Blowfish and SHA(256/512). Warning: This is much slower.
+      },
+      'Author'          =>
+        [
+          'theLightCosine',
+          'hdm',
+          'h00die' # hashcat integration
+        ] ,
+      'License'         => MSF_LICENSE,  # JtR itself is GPLv2, but this wrapper is MSF (BSD)
+      'Actions'         =>
+        [
+          ['john', {'Description' => 'Use John the Ripper'}],
+          ['hashcat', {'Description' => 'Use Hashcat'}],
+        ],
+      'DefaultAction' => 'john',
+    )
+
+    register_options(
+      [
+        OptBool.new('Crypt',[false, 'JtR only: Try crypt (blowfish, sha256/512) format hashes (Very Slow)', false]),
```

Comment:
Let's see if we can remove overlapping options like this and instead filter for JTR at the database layer so we just don't pass down blowfish/sha256/sha512 in the first place to JTR